addons/medical_ai/functions/fnc_healUnit.sqf

```diff
@@ -0,0 +1,84 @@
+/*
+ * Author: BaerMitUmlaut
+ * Makes a medic heal the next unit that needs treatment.
+ *
+ * Arguments:
+ * None
+ *
+ * Return Value:
+ * Nothing
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+// Can't heal other units when unconscious
+if (_this getVariable ["ACE_isUnconscious", false]) exitWith {};
+// Check if we're still treating
+if ((_this getVariable [QGVAR(treatmentOverAt), CBA_missionTime]) > CBA_missionTime) exitWith {};
+
+// Find next unit to treat
+private _healQueue = _this getVariable [QGVAR(healQueue), []];
+private _target = _healQueue select 0;
+
+// If unit died or was healed, be lazy and wait for the next tick
+if (isNull _target || {!alive _target} || {!(_target call FUNC(isInjured))}) exitWith {
+    _target forceSpeed -1;
+    _healQueue deleteAt 0;
+    _this getVariable [QGVAR(healQueue), _healQueue];
+    _this forceSpeed -1;
+    // return to formation instead of going where the injured unit was if it healed itself in the mean time
+    _this doFollow leader _this;
+
+    #ifdef DEBUG_MODE_FULL
+        systemChat format ["%1 finished healing %2", _this, _target];
+    #endif
+};
+
+// Move to target...
+if (_this distance _target > 2) exitWith {
+    if !(_this getVariable [QGVAR(movingToInjured), false]) then {
+        _this setVariable [QGVAR(movingToInjured), true];
+        _this doMove getPosATL _target;
+    };
+};
+_this setVariable [QGVAR(movingToInjured), false];
+
+// ...and make sure medic and target don't move
+_this forceSpeed 0;
+_target forceSpeed 0;
```

Comment:
The medic will only be requested once the injured unit is considered safe, and the medic will only move to the injured unit when the medic is safe. The safety status of the injured unit might change in the mean time, but the medic simply aborts the process when he is not safe anymore.